backend/de.metas.business/src/main/java/de/metas/order/impl/OrderLinePriceCalculator.java

```diff
@@ -504,16 +502,15 @@ private Percent extractDiscount(final IPricingResult pricingResult, final SOTrx
 
 	private boolean isAllowChangingDiscount(@NonNull final SOTrx soTrx)
 	{
-		if (soTrx.isPurchase())
-		{
-			return true;
-		}
-
 		final I_C_OrderLine orderLine = request.getOrderLine();
 		if (orderLine.isManualDiscount())
 		{
 			return false;
 		}
+		if (soTrx.isPurchase())
```

Comment:
i wonder if this shouldn't be after `request.isUpdatePriceEnteredAndDiscountOnlyIfNotAlreadySet()` :thinking: